build_tools/cmake/iree_cc_library.cmake

```diff
@@ -6,6 +6,35 @@
 
 include(CMakeParseArguments)
 
+# iree_filter_cc_deps(DEPS_VAR)
+#
+# Filters a list of CC dependencies, making alterations as needed.
+# At present, this is used to redirect LLVM libraries to the libLLVM.so
+# dynamic library, when LLVM has been configured to link against it.
+# This is necessary to preserve the one-definition rule in the build graph
+# in a consistent way as to how AddLLVM.cmake does it.
+# Note that this only really works if libLLVM.so was configured to contain
+# "all" components. If this ever becomes unworkable, we may need to port the
+# component naming logic and selectively choose when to divert.
+function(iree_filter_cc_deps DEPS_VAR)
+  set(_deps ${${DEPS_VAR}})
+  set(_new_deps)
+  set(_modified FALSE)
+  foreach(_dep ${_deps})
```

Comment:
Maybe this can be `foreach(_dep IN LISTS ${DEPS_VAR})` , so that (1) you never need to actually expand `${${DEPS_VAR}}`, you can let `foreach` do that , and (2) you can drop the `_deps` variable altogether?

Reply:
Also, why is it needed to track `_modified`? Why not unconditionally do the `set(${DEPS_VAR} "${_new_deps}" PARENT_SCOPE)`?

Reply:
Actually, if i'm not mistaken, `list(TRANSFORM)` might replace the whole `foreach`., so the function could be just:

```cmake
function(iree_filter_cc_deps DEPS_VAR)
  if(LLVM_LINK_LLVM_DYLIB)
    list(TRANSFORM ${DEPS_VAR} REPLACE "^LLVM.*" "LLVM")
  endif()
  list(REMOVE_DUPLICATES ${DEPS_VAR})
  set(${DEPS_VAR} "${${DEPS_VAR}}" PARENT_SCOPE)
endfunction()
```